cloudbuild/e2e-test.yaml

```diff
@@ -45,6 +45,7 @@ steps:
       - PIPELINE_TEMPLATE=${_PIPELINE_TEMPLATE}
       - VERTEX_PIPELINE_ROOT=${_TEST_VERTEX_PIPELINE_ROOT}
       - PIPELINE_FILES_GCS_PATH=${_PIPELINE_PUBLISH_GCS_PATH}/${COMMIT_SHA}
+      - ADDITIONAL_SUFFIX=
```

Comment:
would suggest to create the suffix dynamically from [cloud build variables](https://cloud.google.com/build/docs/configuring-builds/substitute-variable-values#using_default_substitutions)

Reply:
I guess there are two scenarios here that we are trying to address:

1. Multiple developers working on pipelines in the same dev project - a user-provided suffix can be used to avoid clashing (as you have done here)
2. Concurrent CI runs in the same project - suffix would need to be autogenerated, but then cleanup becomes an issue

My preference would be to not require the user to provide their own suffix, for example using the pipeline ID or a timestamp of some sort (or git commit hash for CI-triggered runs perhaps)

Clean-up of assets files in GCS is less of an issue as they are not very big (alternatively, you could set up [object lifecycle management](https://cloud.google.com/storage/docs/lifecycle) on the bucket)
Clean-up of bigquery tables could be managed by setting an [expiration time](https://cloud.google.com/bigquery/docs/managing-tables#updating_a_tables_expiration_time) on the table

Reply:
Could also implement a teardown strategy for e2e tests which removes resources which were created as part of the test which addresses (2) well while supporting auto-generated suffixes.

Reply:
I'm onboard with that. For this story, I'll implement the autogenerated suffix approach, and create another story for implementing the clean up 👍

Reply:
@browningjp-datatonic @felix-datatonic  I've been working on the concurrent pipelines capability using an auto-generated suffix and running into a couple of issues

* Generating the suffix in env.sh doesn't work because you can't load it directly into the Makefile. You have to source `env.sh` in the Makefile which would break e2e tests
* Generating the suffx and assigning it to a variable in the Makefile doesn't work because it gets re-generated in the make commands so you'd end up with different values going into `sync-assets` and `compile-pipeline`  commands

The options I've come up with are
1. Combine `sync-assets` and `compile-pipeline` into one command reduces modularity but means we can use one suffix for both
2. We could use arguments in the make commands and pass through the commit hash but it would introduce a lot of if statements depending on whether to use suffix or not for `sync-assets`, `compile-pipeline` and `run` commands
3. Allow users to specify suffix themselves as is already the case in this PR

I think number 3 would be the cleanest and easiest :grimacing: but I may have missed something up to now, what do you think?

Reply:
As decided in the call,  I'll implement the following

 - Use the git username as suffix and store in env.sh as `RESOURCE_SUFFIX` env variable
 - For e2e tests in CI, use the commit hash for the `RESOURCE_SUFFIX` env variable
 - Clean up to be implemented in a later story/PR